### PR TITLE
Expand Build Matrix

### PR DESCRIPTION
**Pull request by jherdman**
Let's also trust Travis to do the right thing with respect to selecting
the OTP version to use.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -27,4 +18,4 @@ sudo: false
 script:
   - mix test
   - mix credo
-  - if [[ `elixir -v` = *"1.6"* ]]; then mix format --check-formatted; fi
+  - if [[ `elixir -v` = *"1.7"* ]]; then mix format --check-formatted; fi
```
Comment (doomspork) [inline .travis.yml]: Why did you change this to 1.7?

File: .travis.yml [hunk 2]
```diff
@@ -1,20 +1,11 @@
 language: elixir
 
 elixir:
+  - 1.7
   - 1.6
   - 1.5
   - 1.4
 
-otp_release:
```
Comment (doomspork) [inline .travis.yml]: Can you link us to the documentation that explains why we don't need this? If you look around pretty much every Elixir project has these OTP releases defined, including Elixir itself.

Comment (jherdman) [reply .travis.yml]: https://docs.travis-ci.com/user/languages/elixir#specifying-otp-release-version I think I'm reading this correctly...

File: .travis.yml (hunk 1 above)
Comment (jherdman) [reply .travis.yml]: 1.7, being the latest public release (outside of 1.8-rc1), felt like the logical choice to run this particular build step on — especially since I'm adding it to the build.

File: .travis.yml (hunk 2 above)
Comment (doomspork) [reply .travis.yml]: I believe so. As I understand it, if we leave this off, it will pick _one_ OTP version. We are intentionally testing against multiples because it is common in real-world situations to have Elixir and OTP on different versions.

File: .travis.yml (hunk 1 above)
Comment (doomspork) [reply .travis.yml]: This is meant to ensure the command isn't run without formatter (versions prior to 1.5).

**Comment by doomspork** [comment]
@jherdman I'm going to close this PR for now. None of these changes will work as expected:

1) We intentionally want to test against multiple OTP versions, not let Travis pick just one.
2) The 1.6 guard is to ensure the formatter is only run for versions in which it exists (so not 1.3, 1.4, and 1.5). It should not be updated with each new release.


Next time feel free to open an issue first to discuss changes not currently documented 👍
